src/screens/Forum/ForumSubcategoryPostsScreen.js

```diff
@@ -44,67 +51,148 @@ export default class ForumSubcategoryPostsScreen extends React.Component {
       }
     });
 
+    const { postLimit } = this.state;
     /* Only query posts w/ categoryID matching categoryID passed in from navigation */
-    this.unsubscribeFromFirestore = firestore().collection('forum_posts')
+    firestore().collection('forum_posts')
       .where('categoryID', '==', categoryID)
       .orderBy('createdAt', 'desc')
-      .onSnapshot((snapshot) => {
+      .limit(postLimit)
+      .get()
+      .then((snapshot) => {
+        // Store data for posts in state
         const forumPosts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
-        this.setState({ forumPosts, loading: false });
-      }, (error) => {
-        this.setState({ errorMessage: error.message, loading: false });
-      });
+        const lastReferencedPost = forumPosts[forumPosts.length - 1];
+
+        this.setState({ forumPosts, lastReferencedPost, loading: false });
+      })
+      .catch((error) => this.setState({ errorMessage: error.message, loadingMore: false }));
   }
 
   componentWillUnmount() {
     this.unsubscribeFromAuth();
-    this.unsubscribeFromFirestore();
   }
 
   navigateToPostScreen(postID, userID) {
     const { navigation } = this.props;
     navigation.navigate('ForumPost', { postID, userID });
   }
 
+  handleEndReached() {
+    const { allPostsLoaded } = this.state;
+
+    if (!allPostsLoaded) {
+      this.loadMore();
+    }
+  }
+
+  // Fetch more data from firestore to load next posts
+  loadMore() {
+    this.setState({ loadingMore: true });
+    const {
+      categoryID, postLimit, forumPosts, lastReferencedPost,
+    } = this.state;
+
+    firestore().collection('forum_posts')
+      .where('categoryID', '==', categoryID)
+      .orderBy('createdAt', 'desc')
+      .startAfter(lastReferencedPost.createdAt)
+      .limit(postLimit)
+      .get()
+      .then((snapshot) => {
+        const newForumPosts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+        const newLastReferenced = newForumPosts[newForumPosts.length - 1];
+
+        if (newForumPosts.length === 0) { // All posts loaded
+          this.setState({ allPostsLoaded: true, loadingMore: false });
+        } else {
+          this.setState({ // Store/append updated data for next postLimit# of posts in state
+            forumPosts: [...forumPosts, ...newForumPosts],
+            lastReferencedPost: newLastReferenced,
+          });
+        }
+      })
+      .catch((error) => this.setState({ errorMessage: error.message, loadingMore: false }));
+  }
+
+  renderHeaderComponent() {
+    const { loading, errorMessage } = this.state;
+
+    if (loading && !errorMessage) {
```

Comment:
Same comment about above with ternary operators

Reply:
(from previous pr) The same issue with ternary operators as above happened, so these were changed as well.